lua/neotest/consumers/state/internal.lua

```diff
@@ -0,0 +1,198 @@
+  if self._status[path] == nil then
+    self._status[path] = {}
+  end
+  self._status[path][name] = status
+end
+
+---Update the cache status count
+function state:_update_cache()
+  for path, results in pairs(self._status) do
+    local count = { passed = 0, failed = 0, skipped = 0, unknown = 0 }
+    for _, status in pairs(results) do
+      count[status] = count[status] + 1
+    end
+    self._cache[path] = count
+  end
+end
+
+---Receive result from event and add to internal state
+---@param adapter_id string
+---@param results table<string, neotest.Result>
+function state:_update_results(adapter_id, results)
+  self._running[string.match(adapter_id, "^[^:]*:(.*)")] = nil
+  self._result[adapter_id] = results
+
+  self:_process_result(results)
+
+  -- Trigger Event
+  vim.api.nvim_exec_autocmds(
+    "User",
+    { pattern = au_result, data = { id = adapter_id, result = results } }
+  )
+end
+
+---Alias for client:get_adapters to get registerd adapters
+---@return string[]
+function state:get_adapters()
+  return self._client:get_adapters()
+end
+
+---Check if there are tests running
+---Optionally, provide a "<adapter_id>:<file_path>" argument to filter on.
+---If no argument is provided, all running processes will be returned.
+---If nil is returned your adapter_id found no match, otherwise an emtpy
+---table is returned.
+---@param adapter_id string optional
+---@param opts table
+---       :fuzzy use string.match instead of direct comparison for key
+---       :as_array if no addapter_id provided, return entire list as array
+---@return table<string, string> | string[] | nil
+function state:running(adapter_id, opts)
+  if adapter_id == nil or (type(adapter_id) ~= string and #adapter_id == 0) then
+    if opts and not opts.as_array then
+      return next(self._running) and self._running or nil
+    end
+
+    local array = {}
+    local i = 1
+    for _, v in pairs(self._running) do
+      array[i] = v
+      i = i + 1
+    end
+    return array
+  end
+
+  for key, value in pairs(self._running) do
+    if
+      (
+        key == adapter_id
+        or (
+          opts
+          and opts.fuzzy
+          and (string.match(adapter_id, escape(key)) or string.match(key, escape(adapter_id)))
+        )
+      ) and next(value) ~= nil
+    then
+      return value
+    end
+  end
+  return nil
+end
+
+---Get back results from cache
+---@param path_query string
+---@param opts table
+---       :fuzzy use string.match instead of direct comparison for key
+---@return table | nil
+function state:get_status(path_query, opts)
+  for key, val in pairs(self._cache) do
+    if key == path_query or (opts and opts.fuzzy and string.match(key, path_query)) then
+      return val
+    end
+  end
+  return nil
+end
+
+---Get status count (passed | failed | skipped | unknown)
+---@param path_query string
+---@param opts table
+---       :fuzzy use string.match instead of direct comparison for key
+---       :status get count for status
```

Comment:
If this is required, it should be it's own parameter and we should assume it's provided